frontend/public/components/utils/k8s-watch-hook.ts

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector, useDispatch } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+import { createSelectorCreator, defaultMemoize } from 'reselect';
+import { makeQuery, makeReduxID } from './k8s-watcher';
+import * as k8sActions from '../../actions/k8s';
+import { K8sResourceKind, K8sKind } from '../../module/k8s';
+import { FirehoseResource } from './types';
+import { RootState } from '../../redux';
+
+export class NoModelError extends Error {
+  constructor() {
+    super('Model does not exist');
+  }
+}
+
+export const useK8sWatch = <R extends K8sResourceKind = K8sResourceKind>(
+  resources: FirehoseResource[] = [],
```

Comment:
What's the reason behind the default value? Is it meaningful to call this hook without an argument?

Reply:
not really, it is for safety reasons. But i guess its not really needed since a caller should make sure that resources are not null. Removed

Reply:
> caller should make sure that resources are not null

Agreed, that's what I meant to say :smiley: